core/primitives/impl/scale_codec_impl.cpp

```diff
@@ -0,0 +1,128 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "scale_codec_impl.hpp"
+
+#include "scale/collection.hpp"
+#include "scale/compact.hpp"
+#include "scale/fixedwidth.hpp"
+
+#include "primitives/block.hpp"
+#include "primitives/scale_error.hpp"  // TODO: change path after moving scale_error to scale
+
+namespace kagome::primitives {
+
+  outcome::result<Buffer> ScaleCodecImpl::encodeBlock(
+      const Block &block) const {
+    Buffer out;
+    OUTCOME_TRY(encoded_header, encodeBlockHeader(block.header()));
+    out.putBuffer(encoded_header);
+
+    // put number of extrinsics
+    bool res = compact::encodeInteger(block.extrinsics().size(), out);
+    if (!res) {
+      std::terminate();
```

Comment:
better use BOOST_ASSERT_MSG(!res, "res is false");

Reply:
ok, I will put it to next commit:
BOOST_ASSERT_MSG(!res, "failed to compact-encode collection size");

Reply:
I mean, though it expands to assert(), it doesn't matter, because this will be removed soon.